src/tests/ports/nfts-rentalsNFTComponent.spec.ts

```diff
@@ -0,0 +1,357 @@
+
+      expect(mockRentalsComponent.getRentalAssets).toHaveBeenCalledWith(
+        expect.objectContaining({
+          first: MAX_SUBGRAPH_QUERY_IN_ELEMENTS,
+        })
+      )
+    })
+  })
+
+  describe('when calling fetch from the marketplaceNFTsComponent', () => {
+    describe('when getRentalAssets has returned one rental asset that is a land', () => {
+      let rentalAssetTokenId: string
+      let rentalAsset: RentalAsset
+
+      beforeEach(() => {
+        rentalAssetTokenId = 'tokenId'
+        rentalAsset = {
+          id: `${mockContractAddresses.land}-${rentalAssetTokenId}`,
+          contractAddress: mockContractAddresses.land,
+          tokenId: rentalAssetTokenId,
+          isClaimed: false,
+          lessor: 'owner',
+        }
+
+        mockRentalsComponent.getRentalAssets = jest
+          .fn()
+          .mockResolvedValue([rentalAsset])
+      })
+
+      it('should call fetch with ids based on the obtained rental asset that is a land', async () => {
+        await rentalsNFTComponent.fetch({})
+
+        expect(mockMarketplaceNFTsComponent.fetch).toHaveBeenCalledWith(
+          expect.objectContaining({
+            ids: [`parcel-${rentalAsset.id}`],
+          })
+        )
+      })
+
+      it('should call fetch without owner in the filters', async () => {
+        await rentalsNFTComponent.fetch({ owner: 'owner' })
+
+        expect(mockMarketplaceNFTsComponent.fetch).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            owner: 'owner',
+          })
+        )
+      })
+    })
+
+    describe('when getRentalAssets has returned one rental asset that is an estate', () => {
+      let rentalAssetTokenId: string
+      let rentalAsset: RentalAsset
+
+      beforeEach(() => {
+        rentalAssetTokenId = 'tokenId'
+        rentalAsset = {
+          id: `${mockContractAddresses.estate}-${rentalAssetTokenId}`,
+          contractAddress: mockContractAddresses.estate,
+          tokenId: rentalAssetTokenId,
+          isClaimed: false,
+          lessor: 'owner',
+        }
+
+        mockRentalsComponent.getRentalAssets = jest
+          .fn()
+          .mockResolvedValue([rentalAsset])
+      })
+
+      it('should call fetch with ids based on the obtained rental asset that is an estate', async () => {
+        await rentalsNFTComponent.fetch({})
+
+        expect(mockMarketplaceNFTsComponent.fetch).toHaveBeenCalledWith(
+          expect.objectContaining({
+            ids: [`estate-${rentalAsset.id}`],
+          })
+        )
+      })
+    })
+  })
+
+  describe('when marketplaceNFTsComponent returns a single nft', () => {
+    let mockNft = { id: 'some-nft' }
+
+    beforeEach(() => {
+      mockMarketplaceNFTsComponent.fetch = jest
+        .fn()
+        .mockResolvedValue([mockNft])
+    })
+
+    it('should finally return an array with that nft', async () => {
+      const result = await rentalsNFTComponent.fetch({})
+
+      expect(result).toEqual([mockNft])
+    })
```

Comment:
If you have a single async function, you can use return and do it in a single line:
```suggestion
    it('should finally return an array with that nft', () => {
     return expect(rentalsNFTComponent.fetch({})).resolves.toEqual([mockNft])
    })
```
Also, you can re-word the test as `should resolve as an array with the NFT` to refer to resolving instead of just returning.